activerecord/lib/active_record/model_schema.rb

```diff
@@ -488,7 +488,10 @@ def load_schema!
           unless table_name
             raise ActiveRecord::TableNotSpecified, "#{self} has no table configured. Set one with #{self}.table_name="
           end
-          @columns_hash = connection.schema_cache.columns_hash(table_name).except(*ignored_columns)
+          
+          columns_hash = connection.schema_cache.columns_hash(table_name)
+          columns_hash = columns_hash.except(*ignored_columns) unless ignored_columns.empty?
```

Comment:
I guess this is a small optimisation to avoid allocating another identical hash?

If `ignored_columns` is empty, we're now freezing a value we don't own - it belongs to the schema cache. We should freeze the value there instead if it's safe to do so, in which case it seems fine to re-freeze it here too.

Reply:
> I guess this is a small optimisation to avoid allocating another identical hash?

Yeah, reading the code I saw that we always create a new hash even though 99% of the time ignore_columns is empty.

>  we're now freezing a value we don't own 

Yes, I need to double check, but I'm pretty sure I froze it in schema cache already.

Reply:
Turns out I didn't. Pretty much all other properties are frozen in schema cache because of deep deduplication, but this one wasn't. I corrected that.